teuthology/suite.py

```diff
@@ -783,23 +783,27 @@ def _build_matrix(path, _isfile=os.path.isfile,
     if _isfile(path):
         if path.endswith('.yaml'):
             return matrix.Base(item)
-        assert False, "Invalid file seen in _build_matrix"
         return None
     if _isdir(path):
+        if path.endswith('.disable'):
+            return None
         files = sorted(_listdir(path))
+        if len(files) == 0:
+            return None
         if '+' in files:
             # concatenate items
             files.remove('+')
             submats = []
             for fn in sorted(files):
-                submats.append(
-                    _build_matrix(
-                        os.path.join(path, fn),
-                        _isfile,
-                        _isdir,
-                        _listdir,
-                        mincyclicity,
-                        fn))
+                submat = _build_matrix(
+                    os.path.join(path, fn),
+                    _isfile,
+                    _isdir,
+                    _listdir,
+                    mincyclicity,
+                    fn)
+                if submat is not None:
```

Comment:
nitpick: why not simply ``if submat:``?  Do we need to allow other falsey values through this condition?